tools/textextractor/TextExtractor/Program.cs

```diff
@@ -0,0 +1,254 @@
+﻿namespace TextExtractor
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+
+    using Windows.Globalization;
+    using Windows.Graphics.Imaging;
+    using Windows.Media.Ocr;
+
+    /// <summary>
+    /// Extracts share count from images using Windows 10 (https://blogs.windows.com/windowsdeveloper/2016/02/08/optical-character-recognition-ocr-for-windows-10/)
+    /// 
+    /// There are two options when running the utility
+    ///     TextExtractor.exe {directory} {outputfile}
+    ///         This syntax will enumerate through all images in a particular directory, analyze the images, and write results for each successful image to a
+    ///         comma-separated file (csv) with the format {fileName},{share count}
+    ///     TextExtractor.exe {file name}
+    ///         This syntax will allow processing of a single file. Output is written to the console. In the case of an error, the exit code will be non-zero and
+    ///         the reason for the failure will be written into the stderr stream.
+    /// </summary>
+    public static class Program
+    {
+        /// <summary>
+        /// The meat of the processing.
+        ///    1) loads an image
+        ///    2) converts it to grayscale (not sure if this is necessary)
+        ///    3) extracts text using windows UWP OCR
+        ///    4) enumerates line by line and parses floats
+        ///    5) does rudimentary logic to determine whether the value is reasonable
+        /// </summary>
+        /// <param name="fileInfo">File to process</param>
+        /// <returns>A tuple with the file name that was processed and the result.</returns>
+        private static async Task<Tuple<string, float>> ProcessSingleImageAsync(FileInfo fileInfo)
+        {
+            Stream fileStream;
+            try
+            {
+                fileStream = File.OpenRead(fileInfo.FullName);
+            }
+            catch(IOException e)
+            {
+                WriteError(fileInfo.Name, "Faile to open file for read", e);
+                throw;
+            }
+
+            using var inputStream = fileStream.AsRandomAccessStream();
+            var decoder = await BitmapDecoder.CreateAsync(inputStream);
+            using var bitmap = await decoder.GetSoftwareBitmapAsync();
+
+            // convert to 8 bit greyscale
+            using var greymap = SoftwareBitmap.Convert(bitmap, BitmapPixelFormat.Gray8);
+
+            var engine = OcrEngine.TryCreateFromLanguage(new Language("en"));
+            var ocrResult = await engine.RecognizeAsync(greymap);
+            var lines = ocrResult.Lines;
+            var candidates = new List<float>();
+        
+            foreach (var item in lines)
+            {
+                if (float.TryParse(item.Text, out var candidate))
```

Comment:
i8n here may cause a problem. Some euroapes have values formatted as `4.200,69` instead of `4,200.69`. Others could be `4 200,69`
Might play with https://docs.microsoft.com/en-us/dotnet/api/system.globalization.numberstyles?view=net-6.0 to see if it's possible to work around this.
https://www.reddit.com/r/Spielstopp/ is a good place to look for images to test with